src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/scenari/EmailGetScenario.scala

```diff
@@ -14,11 +14,14 @@ class EmailGetScenario {
   def generate(duration: Duration, userFeeder : UserFeederBuilder, recipientFeeder: RecipientFeederBuilder): ScenarioBuilder =
     scenario("EmailGetScenario")
       .feed(userFeeder)
-      .exec(CommonSteps.provisionUsersWithMessageList(recipientFeeder, numberOfMessages = 10))
+      .exec(CommonSteps.provisionUsersWithMessages(recipientFeeder, numberOfMessages = 10))
       .exec(SessionStep.retrieveAccountId)
       .during(duration) {
-        exec(JmapEmail.getRandomEmails()
-          .check(JmapHttp.statusOk, JmapHttp.noError, JmapEmail.nonEmptyListMessagesChecks))
+        exec(JmapEmail.queryEmails()
+            .check(JmapHttp.statusOk, JmapHttp.noError, JmapEmail.nonEmptyListMessagesChecks))
```

Comment:
You don't save the email (id) list in the session here?

Reply:
It's done in the `JmapEmail.nonEmptyListMessagesChecks`, that you copied from draft?